### PR TITLE
virts-1752

### PR DESCRIPTION
**Pull request by ArtificialErmine**
## Description

Adds protection against operation cleanup hanging forever by timing out if the agent is misbehaving.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

This has been tested with normal operations running Thief within a AWS environment, and with the automated BRAWLER loop game.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [NA] I have made corresponding changes to the documentation
- [NA] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: app/objects/c_operation.py [hunk 1]
```diff
@@ -63,6 +63,7 @@ def __init__(self, name, agents, adversary, id=None, jitter='2/8', source=None,
         super().__init__()
         self.id = id
         self.start, self.finish = None, None
+        self.timeout = 180
```
Comment (christophert) [inline app/objects/c_operation.py:66]: I think this should be a dynamic configuration option within the operations pane or global application configuration

Comment (ArtificialErmine) [reply app/objects/c_operation.py:66]: Renaming this to be more accurate.

**Review by christophert — APPROVED** [review]

**Review by wbooth — CHANGES_REQUESTED** [review]
Let's make this configurable and add as a separate function for the cleanup/put the try/catch in the wait_for_completion

**Comment by ArtificialErmine** [comment]
> Let's make this configurable and add as a separate function for the cleanup/put the try/catch in the wait_for_completion

I'll make it more configurable, though at the moment it's derived from the configuration of the operation's agents rather than directly by the user. I can pull that try/catch out into a separate function, but I'm going to push back against moving it to _wait_for_completion_, since that will have side effects on operations that aren't configured to auto-close when they run out of agents. At the moment, the override specifically only triggers when cleanup is triggered on an operation, and no other times.

**Pull request merged**